go-controller/pkg/util/multi_network.go

```diff
@@ -939,8 +939,16 @@ func GetPodNADToNetworkMappingWithActiveNetwork(pod *kapi.Pod, nInfo NetInfo, ac
 		Name:      activeNetworkNADKey[1],
 	}
 
+	if AllowsPersistentIPs(nInfo) {
```

Comment:
Do we need to check if this is a primary network here?

Reply:
Why ? Isn't that checked inside the allows persistent IPs helper you've requested ?

Reply:
Right, but the annotation itself is just for primary networks and not secondary ones, right? So if we just use `AllowsPersistentIPs` we would be allowing the use of the annotation for secondary networks as well?

Reply:
Thus, IMHO, coalescing all the info about when persistent ips in a single place is counter-productive. 

... since we'll have to perform redundant checks elsewhere to ensure we're not covering more than what we want.

But I'll blindly throw it in. Done.

Reply:
The way we have the code now, which could be better but I think is an improvement over what we had, highlights that the persistent IPs functionality is the same for all networks and that the only difference is the API. I don't really understand how you see a problem with the code but at the same time be fine with the API differences.

Reply:
The UDN API differences were proposed, discussed, and accepted on an enhancement.

I personally think they do make sense, and simplify the users tasks. Hence I'm OK with it.

Still, I think this is tangential to this particular discussion.